cpp/src/Slice/Parser.cpp

```diff
@@ -954,29 +954,26 @@ Slice::Contained::setMetadata(const list<string>& metadata)
     _metadata = metadata;
 }
 
-FormatType
+optional<FormatType>
```

Comment:
I think we should streamline this format metadata, and drop the compact option.

First, this metadata is only for interface and operations, and the interface metadata (when specified) is "inherited" by all operations. It could be simply:
```
[sliced-format] interface Widget { at least one operation using class otherwise it's pointless }
```
or
```
[format(sliced)] interface MyWidget { ... }
```
However a metadata format with a single mandatory option would be odd.

With this update, this function could become simply:
`bool hasSlicedFormatMedata() const`

We could also consider a further streamlining where this metadata is operation-only, just like oneway and compress.

The default format (when no metadata is specified) is naturally Compact. In C#, it's FormatType.Compact or default.

Reply:
Yes, I see what you mean, going forwards I agree that there any use in keeping the arguments around, since either you don't specify metadata (and get Compact), or you do (and get Sliced).

The issue is that right now is that even though `format:default` and `DefaultFormat` have been dropped from the compiler, they're still used in mappings other than C#.
We can stop building and testing the mappings other than C# and C++, but even in C++ we still have all 3 Default, Compact, and Sliced, which will cause issues.

Reply:
You just need to fix the corresponding language mappings as well.